images/archive/importer.go

```diff
@@ -214,35 +220,108 @@ func onUntarBlob(ctx context.Context, r io.Reader, store content.Ingester, size
 }
 
 func resolveLayers(ctx context.Context, store content.Store, layerFiles []string, blobs map[string]ocispec.Descriptor) ([]ocispec.Descriptor, error) {
-	var layers []ocispec.Descriptor
-	for _, f := range layerFiles {
+	layers := make([]ocispec.Descriptor, len(layerFiles))
+	descs := map[digest.Digest]*ocispec.Descriptor{}
+	filters := []string{}
+	for i, f := range layerFiles {
 		desc, ok := blobs[f]
 		if !ok {
 			return nil, errors.Errorf("layer %q not found", f)
 		}
+		layers[i] = desc
+		descs[desc.Digest] = &layers[i]
+		filters = append(filters, "labels.\"containerd.io/uncompressed\"=="+desc.Digest.String())
+	}
+
+	err := store.Walk(ctx, func(info content.Info) error {
```

Comment:
the filter functionality is TODO item right now.

Reply:
ugh... well as long as this code doesn't panic, just shouldn't assume then that the result is non empty. Maybe an `ok` check on labels would be safer.

Reply:
You are referring to the `local` implementation, the metadata store that will normally be used handles it. Either way, the implementation should always handle the case where the filter wasn't applied.